examples/src/main/kotlin/ai/koog/agents/example/snapshot/FilePersistentAgentExample.kt

```diff
@@ -0,0 +1,129 @@
+    println("Checkpoint directory: $checkpointDir")
+    
+    // Create the file-based checkpoint provider
+    val provider = JVMFileAgentCheckpointStorageProvider(checkpointDir)
+    
+    // Create a unique agent ID to identify this agent's checkpoints
+    val agentId = "persistent-agent-example"
+    
+    // Create tool registry with basic tools
+    val toolRegistry = ToolRegistry {
+        tool(AskUser)
+        tool(SayToUser)
+    }
+
+    // Create agent config with a system prompt
+    val agentConfig = AIAgentConfig(
+        prompt = prompt("persistent-agent") {
+            system("You are a helpful assistant that remembers conversations across sessions.")
+        },
+        model = OllamaModels.Meta.LLAMA_3_2,
+        maxAgentIterations = 20
+    )
+
+    println("Creating and running agent with continuous persistence...")
+    
+    // Create the agent with the file-based checkpoint provider and continuous persistence
+    val agent = AIAgent(
+        promptExecutor = executor,
+        strategy = SnapshotStrategy.strategy,
+        agentConfig = agentConfig,
+        toolRegistry = toolRegistry,
+        id = agentId
+    ) {
+        install(AgentCheckpoint) {
+            snapshotProvider(provider)
+            continuouslyPersistent() // Enable automatic checkpoint creation after each node
+        }
+    }
+
+    // Run the agent with an initial input
+    val result = agent.run("Hello, can you help me with a task?")
+    println("Agent result: $result")
+
+    // Retrieve all checkpoints created during the agent's execution
+    val checkpoints = provider.getCheckpoints(agentId)
+    println("\nRetrieved ${checkpoints.size} checkpoints for agent $agentId")
+
+    // Print checkpoint details
+    checkpoints.forEachIndexed { index, checkpoint ->
+        println("Checkpoint ${index + 1}:")
+        println("  ID: ${checkpoint.checkpointId}")
+        println("  Created at: ${checkpoint.createdAt}")
+        println("  Node ID: ${checkpoint.nodeId}")
+        println("  Message history size: ${checkpoint.messageHistory.size}")
+    }
+
+    // Verify that the checkpoint files exist in the file system
+    val checkpointsDir = checkpointDir.resolve("checkpoints").resolve(agentId)
+    if (checkpointsDir.exists()) {
+        println("\nCheckpoint files in directory: ${checkpointsDir.toFile().list()?.joinToString()}")
+    }
+
+    println("\nNow creating a new agent instance with the same ID to demonstrate restoration...")
+    
+    // Create a new agent instance with the same ID
+    // It will automatically restore from the latest checkpoint
+    val restoredAgent = AIAgent(
+        promptExecutor = executor,
+        strategy = SnapshotStrategy.strategy,
+        agentConfig = agentConfig,
+        toolRegistry = toolRegistry,
+        id = agentId
+    ) {
+        install(AgentCheckpoint) {
+            snapshotProvider(provider)
+            continuouslyPersistent()
+        }
+    }
```

Comment:
I'd rather have some more real-life example like this:

There is a simple conversational agent with automatic persistency storing to some fixed location file, and it gets shut down after, let's say 2 messages (tool calls; note -- or it can be also calculator). This logic can be just in the tool call to just do System.stop() or throw Runtime error if it was called more than 2 times.

And next time you run the same `main()` it should just automagically continue from the same place (either calculating, or chatting with you).